pkg/synopsysctl/cmd_destroy.go

```diff
@@ -74,51 +79,59 @@ var destroyCmd = &cobra.Command{
 }
 
 func destroy(namespace string) error {
-	log.Infof("destroying Synopsys Operator in namespace '%s'...", namespace)
-	crds := []string{}
-	cm, err := util.GetConfigMap(kubeClient, namespace, "synopsys-operator")
-	if err != nil {
-		log.Errorf("unable to find the 'synopsy-operator' config map in namespace '%s' due to %+v", namespace, err)
-	} else {
-		data := cm.Data["config.json"]
-		var cmData map[string]interface{}
-		err = json.Unmarshal([]byte(data), &cmData)
-		if err != nil {
-			log.Errorf("unable to unmarshal config map data due to %+v", err)
-		}
-		if crdNames, ok := cmData["CrdNames"]; ok {
-			crds = strings.Split(crdNames.(string), ",")
-		}
-	}
-
 	// delete namespace
-	checkErr := util.CheckResourceNamespace(restconfig, kubeClient, strings.Join(crds, ","), namespace, true)
-	if err != nil && isForceDestroy {
+	isNamespaceExist, err := util.CheckResourceNamespace(kubeClient, namespace, "", true)
+	if isNamespaceExist && err != nil && isForceDestroy {
```

Comment:
This can be simplified as:

if err != nil {
  if isForceDestory {
    log.Warnf("namespace '%s' cannot be deleted: %v", namespace, err)
  } else {
    return fmt.Errorf("%s. It is not recommended to destroy the Synopsys Operator so these resources can continue to be managed. If you are sure you want to delete the Synopsys Operator anyway then you can use the 'force' option which will keep all the instances and delete only the Synopsys Operator", err.Error())
  }
}

Reply:
done